### PR TITLE
Compute momentum

### PR DESCRIPTION
**Pull request by rupertnash**
**Summary**

Add a compute for the total momentum of a group on atoms, in the style of compute KE

For now, I have added this to the main LAMMPS src/ directory but you may prefer that I contribute this to USER-MISC. If this is the case, please let me know and I will alter the PR to make it so.

This compute is useful for a larger suite of fixes (for coupling between LAMMPS an other codes using MUI https://github.com/MxUI/MUI) I plan to contribute back as a user package in the coming months.

**Related Issues**

**Author(s)**

Rupert Nash, EPCC, The University of Edinburgh, r.nash@epcc.ed.ac.uk

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

No

**Implementation Notes**

I cribbed the class definition from compute_ke.h/.cpp, changing from scalar to vector

This was tested on a variety of input scripts by also dumping atom velocities masses.

No other features are affected.

**Post Submission Checklist**

_Please check the fields below as they are completed **after** the pull request has been submitted. Delete lines that don't apply_

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [x] A package specific README file has been included or updated
- [x] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

**Comment by akohlmey** [comment]
Thanks for putting in the effort to submit a pull request. But why not simply define the following atom style variables:
```
variable mx atom mass*vx
variable my atom mass*vy
variable mz atom mass*vz
```
and then access `v_mx`, `v_my`, and `v_mz` instead of using this compute
```
compute c1 all momentum
```
with access through `c_c1[1]`, `c_c1[2]`, and `c_c1[3]`?
Is there any use case, where using this compute offers significant advantages the atom style variables?

**Comment by rupertnash** [comment]
Thanks for looking at this! However I think you missed that this compute will calc the total momentum of all the atoms in a group whereas (as I understand it and I'm new to LAMMPS...) an atom style variable is per-atom. Is that right?

**Comment by akohlmey** [comment]
> Thanks for looking at this! However I think you missed that this compute will calc the total momentum of all the atoms in a group whereas (as I understand it and I'm new to LAMMPS...) an atom style variable is per-atom. Is that right?

you are correct, I missed that. however, that would just be one more line of input. 

```
variable mx atom mass*vx
variable my atom mass*vy
variable mz atom mass*vz
compute c1 all reduce sum v_mx v_my v_mz
```

**Comment by rupertnash** [comment]
OK! Thanks for the comments Axel. I guess this is a risk with LAMMPS being so big.

However since the code now exists, and would save users some effort/learning, I'm still willing to tick off the remaining items above, if you think it's of value to the users. If you think on balance it's better to use the variable + reduce compute in script then I'm happy if you just close the PR.

**Comment by akohlmey** [comment]
> OK! Thanks for the comments Axel. I guess this is a risk with LAMMPS being so big.
> 
> However since the code now exists, and would save users some effort/learning, I'm still willing to tick off the remaining items above, if you think it's of value to the users. If you think on balance it's better to use the variable + reduce compute in script then I'm happy if you just close the PR.

I am coming out neutral on this one. On one hand, this command allows to do with one line of input script, what otherwise requires four, on the other hand, it is additional code (albeit simple code) to maintain. I was asking for a particular use case for this command versus input script to tip the scales toward including it. If added, it should definitely go into the USER-MISC package, though.

I'll assign the PR to @sjplimp for a second opinion.

File: src/Makefile.list [hunk 1]
```diff
@@ -7,9 +7,9 @@ SHELL = /bin/sh
...
...pp pair.cpp pair_airebo.cpp pair_born_coul_long.cpp pair_buck.cpp pair_buck_coul_cut.cpp pair_buck_coul_long.cpp pair_comb.cpp pair_coul_cut.cpp pair_coul_debye.cpp pair_coul_long.cpp pair_dpd.cpp pair_dpd_tstat.cpp pair_eam.cpp pair_eam_alloy.cpp pair_eam_fs.cpp pair_eim.cpp pair_hybrid.cpp pair_hybrid_overlay.cpp pair_lj96_cut.cpp pair_lj_charmm_coul_charmm.cpp pair_lj_charmm_coul_charmm_implicit.cpp pair_lj_charmm_coul_long.cpp pair_lj_cut.cpp pair_lj_cut_coul_cut.cpp pair_lj_cut_coul_debye.cpp pair_lj_cut_coul_long.cpp pair_lj_cut_coul_long_tip4p.cpp pair_lj_expand.cpp pair_lj_gromacs.cpp pair_lj_gromacs_coul_gromacs.cpp pair_lj_smooth.cpp pair_morse.cpp pair_soft.cpp pair_sw.cpp pair_table.cpp pair_tersoff.cpp pair_tersoff_zbl.cpp pair_yukawa.cpp pppm.cpp pppm_tip4p.cpp random_mars.cpp random_park.cpp read_data.cpp read_restart.cpp region.cpp region_block.cpp region_cone.cpp region_cylinder.cpp region_intersect.cpp region_plane.cpp region_prism.cpp region_sphere.cpp region_union.cpp remap.cpp remap_wrap.cpp replicate.cpp respa.cpp run.cpp set.cpp shell.cpp special.cpp temper.cpp thermo.cpp timer.cpp universe.cpp update.cpp variable.cpp velocity.cpp verlet.cpp write_restart.cpp
```
Comment (akohlmey) [inline src/Makefile.list]: no need to make modifications to this file. this needs to be re-generated via `make list` based on installed packages anyway, and there are not that many broken make programs around.

Comment (rupertnash) [reply src/Makefile.list]: Thanks - reverted in commit just pushed

**Comment by sjplimp** [comment]
I think it's fine to add to USER-MISC as a new compute.

**Comment by rupertnash** [comment]
In the last few commits I've moved the compute to USER-MISC, updated the docs, and added a test input script. I think this is now ready to go.

**Comment by sjplimp** [comment]
agreed: ready to go when you make the doc page edits.  Thanks

**Comment by rupertnash** [comment]
Sorry @sjplimp - I don't follow which doc page needs updated. I've done
- doc/src/compute.txt
- doc/src/Commands_compute.txt
- doc/src/compute_momentum.txt (note about USER-MISC in restrictions similar to e.g. compute_entropy_atom)
- doc/src/computes.txt
- src/USER-MISC/README
Let me know what I've missed!
Cheers

**Comment by sjplimp** [comment]
I just meant the paragraph at the bottom of compute_momentum.txt that I flagged
in an earlier comment.  It has some wrong grammar.

**Comment by rupertnash** [comment]
OK - I think I've got the grammar error- afraid I didn't see anything explicitly flagged. Also resolved the merge conflicts. Thanks.

**Comment by akohlmey** [comment]
@sjplimp this also seems to be ready and waiting to be approved and assigned to me

**Pull request merged**